packages/cli/src/util/client.ts

```diff
@@ -20,31 +25,40 @@ export async function createMedplumClient(options: MedplumClientOptions, profile
     medplumClient.setAccessToken(accessToken);
   }
 
-  const clientId = options.clientId || process.env['MEDPLUM_CLIENT_ID'];
-  const clientSecret = options.clientSecret || process.env['MEDPLUM_CLIENT_SECRET'];
-
   if (clientId && clientSecret) {
     medplumClient.setBasicAuth(clientId, clientSecret);
     await medplumClient.startClientLogin(clientId, clientSecret);
   }
   return medplumClient;
 }
 
-function getClientValues(options: MedplumClientOptions, storage: FileSystemStorage): Record<string, any> {
+function getClientValues(options: MedplumClientOptions, storage: FileSystemStorage): MedplumClientOptions {
+  const storageOptions: any = storage.getObject('options');
   const baseUrl =
-    options.baseUrl ?? storage.getString('baseUrl') ?? process.env['MEDPLUM_BASE_URL'] ?? 'https://api.medplum.com/';
-  const fhirUrlPath =
-    options.fhirUrlPath ?? storage.getString('fhirUrlPath') ?? process.env['MEDPLUM_FHIR_URL_PATH'] ?? '';
+    options.baseUrl ?? storageOptions?.baseUrl ?? process.env['MEDPLUM_BASE_URL'] ?? 'https://api.medplum.com/';
+  const fhirUrlPath = options.fhirUrlPath ?? storageOptions?.fhirUrlPath ?? process.env['MEDPLUM_FHIR_URL_PATH'] ?? '';
   const accessToken =
-    options.accessToken ?? storage.getString('accessToken') ?? process.env['MEDPLUM_CLIENT_ACCESS_TOKEN'] ?? '';
-  const tokenUrl = options.tokenUrl ?? storage.getString('tokenUrl') ?? process.env['MEDPLUM_TOKEN_URL'] ?? '';
+    options.accessToken ?? storageOptions?.accessToken ?? process.env['MEDPLUM_CLIENT_ACCESS_TOKEN'] ?? '';
+  const tokenUrl = options.tokenUrl ?? storageOptions?.tokenUrl ?? process.env['MEDPLUM_TOKEN_URL'] ?? '';
   const authorizeUrl =
-    options.authorizeUrl ?? storage.getString('authorizeUrl') ?? process.env['MEDPLUM_AUTHORIZE_URL'] ?? '';
-  const fetchApi = options.fetch ?? fetch;
+    options.authorizeUrl ?? storageOptions?.authorizeUrl ?? process.env['MEDPLUM_AUTHORIZE_URL'] ?? '';
+
+  const clientId = options.clientId ?? storageOptions?.clientId ?? process.env['MEDPLUM_CLIENT_ID'];
+  const clientSecret = options.clientSecret ?? storageOptions?.clientSecret ?? process.env['MEDPLUM_CLIENT_SECRET'];
 
-  return { baseUrl, fhirUrlPath, accessToken, tokenUrl, authorizeUrl, fetchApi };
+  return { baseUrl, fhirUrlPath, accessToken, tokenUrl, authorizeUrl, clientId, clientSecret };
 }
 
 export function onUnauthenticated(): void {
   console.log('Unauthenticated: run `npx medplum login` to sign in');
 }
+
+function checkForProfile(storage: FileSystemStorage, profile: string): void {
+  if (profile === 'default') {
```

Comment:
Put the check for profile here, and skip if the profile is 'default'

Reply:
Yep, this works 👍